pkg/storage/registry/spaces/spaces_test.go

```diff
@@ -333,24 +333,24 @@ var _ = Describe("Spaces", func() {
 
 			It("filters by id", func() {
 				filters := map[string]string{
-					"storage_id": "barspace",
-					"opaque_id":  "barspace",
+					"storage_id": "foospace",
+					"opaque_id":  "foospace",
```

Comment:
hm, `filters by id` should only test if filtering by id works. changing from `barspace` to `foospace` only affects the tests because a different provider is used. And the username lowercasing is the last change. How is this testing the `mount_point` logic you changed? Should it?

Could you update a more specific test of it it does not exist add one that covers your changes?

That being said, I noticed that the test sets up two `personal` spaces, both owned by alice. That should not be the case: either a filter by type `personal` should list all personal spaces and each is owned by a different user, or a filter by owner returns the storage provider that is responsible for that specific space.

We'll look into that.

Reply:
What I mean to say is a storage provider with the config:
```
"mount_point":   "/users/[l-z]"
```
should not return a space with path `/users/Alice`

To make the test pass, I just changed the resource ID to one of a provider capable of returning such a path. I can add one more test where sending a request with a `barspace` filter should return empty.

Regarding the capitalization, again, with a `mount_point: /users/[a-k]`, you shouldn't return `/users/Alice`

Reply:
The `mount_point` is used for path based access in a sharded directory. It should determine which storage provider should be queried for a given _path_. When querying by id or with any other filter every storage provider needs to be queried (unless we can take into account the storageproviderid, eg. for an id based request).

The list of spaces returend by a storage provider should never be filtered by `mount_point`:
- either the path determined the storage provider, based on the mount_point or
- the id determined the provider and the path is built by the `path_template`

Reply:
But that's just inconsistent. The path built by `path_template` should always be of the form dictated by `mount_point`.

Assume that the directory `/users/[username]` exists in both the sharded providers but according to our sharding config, we only provide a user access to one. Both of this would be returned when the user queries personal spaces.
```
[
  {"localhost:11001": {"/users/alice", ...}},
  {"localhost:11002": {"/users/alice", ...}},
]
```

Config:
```
/users/[a-k]: "localhost:11001",
/users/[l-z]: "localhost:11002",
```

Reply:
So we can add a log, not return this response to the user

Reply:
This would also hamper the case when you want to balance load among users based on their usernames, even though the reva daemons might be connecting to the same storage backend.